packages/discussions/src/types.ts

```diff
@@ -705,6 +705,55 @@ export interface ISearchChannelsOptions extends IHubRequestOptions {
   params?: ISearchChannels;
 }
 
+/**
+ * Role types
+ *
+ * @export
+ * @enum {string}
+ */
+export enum Role {
+  READ = "read",
+  WRITE = "write",
+  READWRITE = "readWrite",
+  MODERATE = "moderate",
+  MANAGE = "manage",
+  OWNER = "owner",
+}
+
+/**
+ * permission object that will populate ACL interface
+ *
+ * @export
+ * @interface IPermission
+ */
+export interface IPermission {
+  role: Role;
+  createdAt: string;
+  updatedAt: string;
+  accessibleAfter: string;
+}
+
+/**
+ * request options for creating channel with ACL
+ *
+ * @export
+ * @interface IWithACL
+ * @extends {ICreateChannelWithACL}
+ */
+export interface ICreateChannelWithACL extends IWithSettings {
+  anonymous?: IPermission;
```

Comment:
Is there a scenario where the client would call `createChannel` and provide `createdAt` and `updatedAt` values for the `authentication`, `groups`, `orgs`, and `users` permission? Seems like that's something that would be included in the output, i.e. `IChannel`, but not necessarily `ICreateChannelWithACL`. If that's true, we could use the [omit utility](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) to define another type, e.g.

```
export interface IPermissionOptions extends Omit<IPermission, "createdAt", "updatedAt">

export interface ICreateChannelWithACL extends IWithSettings {
  anonymous?: IPermissionOptions;
  // ...
}
```

I believe we also need to make sure that `IChannel` reflects the channel schema returned when calling the `getById` and `search` channel endpoints. When I pull the release for this PR merge, accessing `channel.acl` results in the typescript error `Property 'acl' does not exist on type 'IChannel'.`

Reply:
@rweber-esri So a couple of things here:

1. The scenario at the moment does involve the client to provide `createdAt` and `updatedAt` for those entities that you mentioned. In the Discussions API, I believe we need to add a way for those fields to be populated with a date/time of current value in some way rather than making the client enter that in. 

2. I see your concern with the `IChannel` needing to reflect the schema of a channel. I believe it is related to this interface
```
export interface IChannel
  extends IWithSettings,
    IWithSharing,
    IWithAuthor,
    IWithEditor,
    IWithTimestamps {
  id: string;
}
```
I didn't change this during my PR and I see now that `acl` is not covered here. I wonder if I need to add an acl object in here like the `id` but make it the type of the interface `ICreateChannelWithACL`

Reply:
> 1. The scenario at the moment does involve the client to provide `createdAt` and `updatedAt` for those entities that you mentioned. In the Discussions API, I believe we need to add a way for those fields to be populated with a date/time of current value in some way rather than making the client enter that in.

ha, that's really remarkable...

Reply:
I said this before, but a lot of the confusion here is that we are trying to use the same interfaces for requesting a channel to be created _and_ the shape of a channel that gets returned.  They share some common properties, but they are different things.